src/main/webapp/query-control.js

```diff
@@ -216,6 +224,75 @@ async function createStateDropdownList() {
   });
 }
 
+// Set up an autocomplete dropdown, attached to the main location input
+// and states dropdown, to suggest places to users as they type
+function setupAutocompleteLocation() {
+  const autocompleteInput = document.getElementById('location-list');
+  const autocompleteResults = document.getElementById('location');
+  const service = new google.maps.places.AutocompleteService();
+  const defaultLocationOptions = autocompleteResults.innerHTML; // all states
+
+  function displaySuggestions(predictions, status) {
+    if (status != google.maps.places.PlacesServiceStatus.OK) {
+      alert(
+          'There was an error while trying to generate location options: ' +
+          status);
+      return;
+    }
+    if (predictions === null) { // No predictions generated for this location
+      return;
+    }
+
+    // Each time we display suggestions, show only the top 5, and then the
+    // normal state dropdown afterwards
+    predictions.splice(5);
+    const resultsHtml = [];
+    predictions.forEach(function(prediction) {
+      resultsHtml.push(
+          '<option class="autocomplete-item" value="' +
+          prediction.description +
+          '"></option>');
+    });
+    autocompleteResults.innerHTML =
+        resultsHtml.join('') + defaultLocationOptions;
+  };
+
+  // When the input box changes (due to typing), get what has been typed and
+  // send it to the autocomplete service, which then calls displaySuggestions
+  // on the predictions it generated
+  autocompleteInput.addEventListener('input', debounce(function() {
+    const value = this.value;
+    value.replace('"', '\\"').replace(/^\s+|\s+$/g, ''); // Trim whitespace
+    service.getPlacePredictions({
+      'input': value,
+      'types': ['(regions)'], // no businesses, just cities counties etc.
+      'componentRestrictions': {'country': 'us'}, // USA only
+    }, displaySuggestions);
+  }, 250)); // wait 250 ms before sending the request so they don't overload
```

Comment:
I think a slightly shorter delay may be better, so users can see available suggestions before continuing typing. I also tried this without the delay locally and it seems to work fine with no delay, what issues did you run in to without the delay? I do notice that the shake becomes more obvious.

Reply:
Yeah, I added the delay to try to get rid of the shake. Maybe I'm a bit sensitive but for me it makes it pretty hard to read the results list when it's at that level of shaking

Reply:
Tried out some options - 175 or 200 seem like they might be the sweet spot here for shaking vs too unresponsive.